actors/builtin/power/power_actor.go

```diff
@@ -214,106 +212,26 @@ func (a Actor) OnSectorProveCommit(rt Runtime, params *OnSectorProveCommitParams
 	rt.ValidateImmediateCallerType(builtin.StorageMinerActorCodeID)
-	builtin.RequireParam(rt, params.RawBytePower.GreaterThanEqual(big.Zero()), "negative raw power %s", params.RawBytePower)
-	builtin.RequireParam(rt, params.QualityAdjustedPower.GreaterThanEqual(big.Zero()), "negative QA power %s", params.RawBytePower)
 	minerAddr := rt.Message().Caller()
 
 	var st State
 	rt.State().Transaction(&st, func() interface{} {
-		err := st.AddToClaim(adt.AsStore(rt), minerAddr, params.RawBytePower.Neg(), params.QualityAdjustedPower.Neg())
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to remove power raw %s, qa %s", params.RawBytePower, params.QualityAdjustedPower)
+		err := st.AddToClaim(adt.AsStore(rt), minerAddr, params.RawByteDelta, params.QualityAdjustedDelta)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to update power raw %s, qa %s", params.RawByteDelta, params.QualityAdjustedDelta)
 		return nil
 	})
 	return nil
 }
 
-type OnFaultEndParams struct {
-	RawBytePower         abi.StoragePower
-	QualityAdjustedPower abi.StoragePower
-}
-
-// Removes power for the calling actor.
-// May only be invoked by a miner actor.
-// TODO: remove duplication of OnSectorTerminate, https://github.com/filecoin-project/specs-actors/issues/487
-func (a Actor) OnFaultEnd(rt Runtime, params *OnFaultEndParams) *adt.EmptyValue {
-	rt.ValidateImmediateCallerType(builtin.StorageMinerActorCodeID)
-	builtin.RequireParam(rt, params.RawBytePower.GreaterThanEqual(big.Zero()), "negative raw power %s", params.RawBytePower)
-	builtin.RequireParam(rt, params.QualityAdjustedPower.GreaterThanEqual(big.Zero()), "negative QA power %s", params.RawBytePower)
-	minerAddr := rt.Message().Caller()
-
-	var st State
-	rt.State().Transaction(&st, func() interface{} {
-		err := st.AddToClaim(adt.AsStore(rt), minerAddr, params.RawBytePower, params.QualityAdjustedPower)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add power raw %s, qa %s", params.RawBytePower, params.QualityAdjustedPower)
-		return nil
-	})
-	return nil
-}
-
-type OnSectorModifyWeightDescParams struct {
-	PrevRawBytePower         abi.StoragePower
-	NewRawBytePower          abi.StoragePower
-	PrevQualityAdjustedPower abi.StoragePower
-	NewQualityAdjustedPower  abi.StoragePower
-}
-
-// Removes power and replaces for the calling actor.
-// May only be invoked by a miner actor.
-// Returns new initial pledge, now committed in place of the old.
-// This is just a composition of OnSectorTerminate and OnSectorProveCommit.
-// TODO: remove duplication of OnSectorTerminate, https://github.com/filecoin-project/specs-actors/issues/487
-func (a Actor) OnSectorModifyWeightDesc(rt Runtime, params *OnSectorModifyWeightDescParams) *abi.TokenAmount {
-	rt.ValidateImmediateCallerType(builtin.StorageMinerActorCodeID)
-	builtin.RequireParam(rt, params.PrevRawBytePower.GreaterThanEqual(big.Zero()), "negative prev raw power %s", params.PrevRawBytePower)
```

Comment:
Have these checks (and the others removed from other methods) become deprecated?  Were they never useful?  Is there anywhere we should consider moving them?

Reply:
I added them only the the previous PR. Previously there were different methods for adding xor removing power for different scenarios. Now we're down to just two, and the new `UpdateClaimedPower` is intended to handle both positive and negative deltas, without being opinionated about why it's being called.